### PR TITLE
[PKG-3347] Update to v2.14.2

### PR DESCRIPTION
**Pull request by ViridianMelody**
[PKG-3347](https://anaconda.atlassian.net/browse/PKG-3347)
[Upstream repository](https://github.com/plotly/dash/tree/v2.14.2)

Target channel: Defaults

- Tests skipped because they require `percy-selenium-python`

[PKG-3347]: https://anaconda.atlassian.net/browse/PKG-3347?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ source:
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
   skip: true  # [py<36]
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: if we removed
```
- importlib-metadata ==4.8.3  # [py<37]
- contextvars ==2.4  # [py<37]

```   
then I think we need to skip `py<38`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -15,8 +15,8 @@ source:
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
-  skip: true  # [py<36]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  skip: true  # [py<38]
```
Comment (cbouss) [inline recipe/meta.yaml:20]: entry points may be added:
https://github.com/plotly/dash/blob/v2.14.2/setup.py#L39

File: recipe/meta.yaml [hunk 3]
```diff
@@ -29,17 +29,22 @@ requirements:
...
+    - requests
+    - retrying
+    - ansi2html
+    - nest-asyncio
     # compress extra
     - flask-compress
   run_constrained:
     - dash-core-components ==2.0.0
     - dash-html-components ==2.0.0
     - dash-table ==5.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:47]: More run
```suggestion
    - dash-table ==5.0.0
    - diskcache >=5.2.1
    - multiprocess >=0.70.12
    - psutil >=5.8.0
    - redis >=3.5.3
    - celery >=5.1.2
    - importlib-metadata <5 # [py<38]
```
https://github.com/plotly/dash/blob/v2.14.2/requires-diskcache.txt
https://github.com/plotly/dash/blob/v2.14.2/requires-celery.txt
https://github.com/plotly/dash/blob/v2.14.2/setup.py#L34-L35

File: recipe/meta.yaml [hunk 4]
```diff
@@ -29,17 +29,22 @@ requirements:
...
+    - importlib-metadata
+    - typing_extensions >=4.1.1
+    - requests
+    - retrying
+    - ansi2html
+    - nest-asyncio
     # compress extra
     - flask-compress
   run_constrained:
     - dash-core-components ==2.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:45]: dash- packages should be in the run section, unless there is a good reason not to:
https://github.com/plotly/dash/blob/v2.14.2/requires-install.txt#L4-L6
https://github.com/plotly/dash/blob/v2.14.2/setup.py#L28

File: recipe/meta.yaml (hunk 3 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:47]: The celery and diskcache requirements are optional dependencies for extras, but I could add them to run_constrained. `flask-compress` is also optional, but it's been included in past versions, so I left it in.

File: recipe/meta.yaml (hunk 4 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:45]: These components are now built into `dash`. The separate packages for them exist only for backwards compatibility.

https://github.com/plotly/dash-core-components
https://github.com/plotly/dash-html-components
https://github.com/plotly/dash-table

File: recipe/meta.yaml (hunk 2 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:20]: Unless I'm mistaken, the `console_scripts` run specific portions of the build process, so it doesn't make sense to distribute them with a package we've already built. The `pytest11` entry point is used to run tests that we're missing dependencies for.

**Review by cbouss — APPROVED** [review]
Nice work!

**Pull request merged**